### ISSUE TITLE
bug: vue 3 with vite pages dont change page

### ISSUE THREAD

**Issue by productdevbook** [ionic-team/ionic-framework#26276]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

https://user-images.githubusercontent.com/38668796/201534856-152e0fd9-b669-49bc-92a7-be317b250f1f.mp4

### Expected Behavior

[Delete](https://github.com/productdevbook/bug-ionic-vite-vue/blob/main/src/pages/%5Busername%5D.vue) vue and fixed. But how to with used, [vite-pages](https://github.com/hannoeru/vite-plugin-pages).

 
https://user-images.githubusercontent.com/38668796/201534941-436c3f5f-e290-4782-8b3e-724783d01b04.mp4

### Steps to Reproduce

1. pnpm install
2. pnpm dev
3. click "check" button
4. click "check settings" button
5. click "go user profile settings" button
6. click "Logout Button" button

### Code Reproduction URL

https://github.com/productdevbook/bug-ionic-vite-vue

### Ionic Info

Ionic:

   Ionic CLI : 6.20.0

Utility:

   cordova-res : not installed globally
   native-run  : not installed globally

System:

   NodeJS : v16.18.1
   npm    : 9.1.1
   OS     : Linux 5.17

**Comment by liamdebeasi** [comment]
Thanks for the report. I can reproduce this issue, but this does not appear to be a bug in Ionic. The problem here is that you are rendering additional `IonRouterOutlet` and `IonTabs` instances when you do not need to.

In general, there should really only be 1 tabs instance and 2 router outlets in a tabs-based application. This issue is likely related to how the `default.vue` template was constructed to always render `IonTabs` and an additional `IonRouterOutlet`.

The pages for the "/" and "auth-check" routes should not be in `IonTabs` nor should they be in a nested `IonRouterOutlet` instance. As noted in the [Ionic Vue Documentation](https://ionicframework.com/docs/vue/navigation#nested-routes) nested routes should only be used when you want to render content in multiple outlets. For these two pages, you are only rendering content in a single outlet.

The page for the "/:username" route should only render a single `IonRouterOutlet` instance inside of `IonTabs`. Currently your application renders two nested `IonRouterOutlet` instances inside of `IonTabs`.

Fixing these two issues should resolve the problem.

**Comment by productdevbook** [comment]
Thans for coming back, Can you play on the repo I sent and send a pr so that I can understand it ? Because I still don't understand how it would be possible with this structure. Sorry :(

**Comment by liamdebeasi** [comment]
Unfortunately, I am not able to submit a PR for your application. However, I can give you some pointers for how to address issue this in your application:

1. I recommend avoiding the automatic template/page generation until you have a better sense of how your application should be structured. This may mean some template duplication in the short term, but I think it will lead to a better understanding of both Ionic and the Vue Router long term.
2. I recommend updating your routes to look something like this (note: this is pseudo code, so you may need to tweak this to get it to compile):

```javascript
const routes = [
  {
    path: '/',
    component: LoginPageComponent
  },
  {
    path: '/auth/check',
    component: AuthCheckComponent
  },
  {
    path: '/dashboard/:username',
    component: TabsComponent,
    children: [
      {
        path: '',
        redirect: '/dashboard/:username/home'
      },
      {
        path: 'home',
        component: HomeTabComponent
      },
      {
        path: 'settings',
        component: SettingsTabComponent
      }
    ]
  }
]
```
3. Do not put `IonTabs` or `IonRouterOutlet` instances into your `LoginPageComponent`, `AuthCheckComponent`, `HomeTabComponent`, or `SettingsTabComponent` files as they are not needed.
4. Put one `IonTabs` instance in `TabsComponent`. Be sure to include an `IonRouterOutlet` and an `IonTabBar`. You can see https://github.com/ionic-team/starters/blob/main/vue/official/tabs/src/views/TabsPage.vue for an example of how we do this in an Ionic Vue starter application.

To summarize, only `TabsComponent` should have `IonTabs` with an `IonRouterOutlet`. I am going to close this as this is not a bug in Ionic, but let me know if you have any additional questions and I will be happy to answer them.
